### PR TITLE
Add burnham test report table

### PR DESCRIPTION
**Pull request by jklukas**
For https://github.com/mozilla/burnham/issues/39

### CODE REVIEW

File: sql/burnham_derived/test_result_v1/init.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE
+  burnham_derived.test_result_v1(
+    submission_timestamp TIMESTAMP,
+    test_run STRING,
+    test_name STRING,
+    test_outcome STRING,
+    test_duration_seconds FLOAT64,
```
Comment (jklukas) [inline sql/burnham_derived/test_result_v1/init.sql:7]: Floats can be a bit of a pain to work with, particularly when converting between JSON and other formats. Can we consider changing this to an INT64 called `test_duration_millis` or `test_duration_micros`?

**Comment by jklukas** [comment]
I've gone ahead and created this table, so you should be able to see it in BQ. If we make some changes during review, I'll recreate before merging.

File: sql/burnham_derived/test_result_v1/init.sql (hunk 1 above)
Comment (hackebrot) [reply sql/burnham_derived/test_result_v1/init.sql:7]: Yes, we can change this to an `INT64` called `test_duration_millis` or `test_duration_milliseconds`.

The most recent test run took `4 passed in 3.82s`. That granularity is good enough!

File: sql/burnham_derived/test_result_v1/schema.bq.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+[
+  {
+    "name": "submission_timestamp",
+    "type": "TIMESTAMP",
+    "description": "Time at which this test report was submitted"
```
Comment (hackebrot) [inline sql/burnham_derived/test_result_v1/schema.bq.json:5]: Does BigQuery write this field automatically?

File: sql/burnham_derived/test_result_v1/schema.bq.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "name": "test_duration_seconds",
+    "type": "FLOAT64",
+    "description": "Duration of the test run"
+  },
+  {
+    "name": "test_report",
+    "type": "STRING",
+    "description": "Markdown-formatted report for error or failed tests"
+  }
+]
```
Comment (hackebrot) [inline sql/burnham_derived/test_result_v1/schema.bq.json:32]: As discussed on https://github.com/mozilla/burnham/issues/39#issuecomment-670516119 we also need a `test_log_url` field that contains a URL to the Airflow job.

File: sql/burnham_derived/test_result_v1/init.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
+CREATE OR REPLACE TABLE
+  burnham_derived.test_result_v1(
```
Comment (hackebrot) [inline sql/burnham_derived/test_result_v1/init.sql:2]: Can we rename the table to `test_results_v1`?

File: sql/burnham_derived/test_result_v1/schema.bq.json (hunk 2 above)
Comment (jklukas) [reply sql/burnham_derived/test_result_v1/schema.bq.json:5]: No, we should add this in the DAG/burnham logic. It should be an ISO string like "2020-01-01 12:00:00.000" in the json we output.

File: sql/burnham_derived/test_results_v1/schema.bq.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  {
+    "name": "test_run",
+    "type": "STRING",
+    "description": "ID of the test run"
+  },
+  {
+    "name": "test_name",
+    "type": "STRING",
+    "description": "Name of the test run"
```
Comment (hackebrot) [inline sql/burnham_derived/test_results_v1/schema.bq.json:15]:
```suggestion
    "description": "Name of the test scenario"
```

File: sql/burnham_derived/test_results_v1/schema.bq.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  {
+    "name": "test_outcome",
+    "type": "STRING",
+    "description": "Outcome of the test run; one of ERROR, FAILED, PASSED, SKIPPED, XFAILED, XPASSED"
```
Comment (hackebrot) [inline sql/burnham_derived/test_results_v1/schema.bq.json:20]:
```suggestion
    "description": "Outcome of the test scenario; one of ERROR, FAILED, PASSED, SKIPPED, XFAILED, XPASSED"
```

File: sql/burnham_derived/test_results_v1/schema.bq.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  {
+    "name": "test_duration_millis",
+    "type": "INT64",
+    "description": "Duration of the test run"
```
Comment (hackebrot) [inline sql/burnham_derived/test_results_v1/schema.bq.json:25]:
```suggestion
    "description": "Duration of the test scenario"
```

**Pull request merged**